### PR TITLE
Implements medipacks & equipment

### PR DESCRIPTION
**Pull request by Ketrai**
## About The Pull Request

Adds medipacks
- Heals health in all health categories.
- Based off damage taken.
- twice as long taken to apply to self

Small
- Heals 12.5 flat + 5%
- 1 second to apply base.
- fits in pockets.

Medium
- heals 9 flat + 20%
- 1.25 seconds to apply base.
- fits in backpacks.
- comes with 1 slime jelly for a small heal after the application.

Large
- heals 5 flat + 45%
- 2 seconds to apply base.
- Takes 10 times as long (20 seconds) To apply to self. Much harder to use alone.
- doesn't fit in backpacks without a specialized holder.
- comes with 7.5 slime jelly for a large heal after application.

Adrenaline
- Can revive the dead.
- Heavily reduced effectiveness on living people, unless you need a quick blood top up.
- heals the dead quite a bit, but you may need multiple syringes to heal someone heavily damaged.
- 20 seconds to apply on the dead.
- 1 second to apply to the living base.
- fits in pockets.

## Why It's Good For The Game

Dead space universe stuff for healing!

## Changelog

:cl:
add: Adds medipacks, use them for a quick heal!
add: Adds medipack holders, use them for storing large medipacks in backpacks.
add: Adds adrenaline medipacks. Use them to revive the dead!
/:cl:

### CODE REVIEW

File: code/modules/reagents/reagent_containers/medipack.dm [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+/obj/item/storage/fancy/medipack/update_overlays()
+	. = ..()
+	if(!is_open)
+		return
+
+	var/medipacks = 0
+	for(var/_medipack in contents)
+		var/obj/item/medipack/large/medipack = _medipack
+		if (!istype(medipack))
+			continue
```
Comment (Kapu1178) [inline code/modules/reagents/reagent_containers/medipack.dm:144]: wtf
```suggestion
	var/medipacks = 0

	for(var/obj/item/medipack/large/medipack in contents)
```

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -810,6 +810,35 @@
 
 	return stat != DEAD
 
+//proc used to heal a mob, but only damage types specified.
+/mob/living/proc/specific_heal(blood_amt = 0, brute_amt = 0, fire_amt = 0, tox_amt = 0, oxy_amt = 0, clone_amt = 0, organ_amt = 0, stam_amt = 0, specific_revive = FALSE, specific_bones = FALSE)
+	if(iscarbon(src))
```
Comment (Kapu1178) [inline code/modules/mob/living/living.dm:815]: as on discord, use proper OOP
image

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/living.dm (hunk 2 above)
Comment (Ketrai) [reply code/modules/mob/living/living.dm]: I think this is fixed?

File: code/modules/reagents/reagent_containers/medipack.dm (hunk 1 above)
Comment (Ketrai) [reply code/modules/reagents/reagent_containers/medipack.dm]: Donezo

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -810,6 +810,38 @@
...
+	adjustBruteLoss(brute_amt*-1)
+	adjustFireLoss(fire_amt*-1)
+	adjustToxLoss(tox_amt*-1)
+	adjustOxyLoss(oxy_amt*-1)
+	adjustCloneLoss(clone_amt*-1)
+	adjustStaminaLoss(stam_amt*-1)
+
+/mob/living/carbon/specific_heal(blood_amt = 0, brute_amt = 0, fire_amt = 0, tox_amt = 0, oxy_amt = 0, clone_amt = 0, organ_amt = 0, stam_amt = 0, specific_revive = FALSE, specific_bones = FALSE)
+	var/mob/living/carbon/C = src
+	if(!(C.dna?.species && (NOBLOOD in C.dna.species.species_traits)))
```
Comment (DTraitor) [inline code/modules/mob/living/living.dm:828]:
```suggestion
	if(!(NOBLOOD in C.dna?.species?.species_traits))
```

File: code/modules/reagents/reagent_containers/medipack.dm [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+	var/mob/living/H = target
+	if(H.stat == DEAD && life_loss)
+		heal_modifier = death_heal
+		heal_delay = 200 //Reviving the dead takes a while, 20 seconds to be exact
+		organ_loss = 30
+		to_chat(user, "<span class='warning'>You begin using the [src] to try and bring [H] back from the dead...</span>")
+	if(H.stat == DEAD && !life_loss) // Won't revive the dead, except for specific extracts
+		to_chat(user, "<span class='warning'>[src] will not work on the dead!</span>")
+		return
+	else if(H.stat != DEAD && life_loss) // Syringes healing the dead won't work well on the living.
```
Comment (DTraitor) [inline code/modules/reagents/reagent_containers/medipack.dm:42]:
```suggestion
	if(H.stat == DEAD)
		if(life_loss)
			heal_modifier = death_heal
			heal_delay = 200 //Reviving the dead takes a while, 20 seconds to be exact
			organ_loss = 30
			to_chat(user, "<span class='warning'>You begin using the [src] to try and bring [H] back from the dead...</span>")
		else // Won't revive the dead, except for specific extracts
			to_chat(user, "<span class='warning'>[src] will not work on the dead!</span>")
			return
	else if(life_loss) // Syringes healing the dead won't work well on the living.
```

File: code/modules/reagents/reagent_containers/medipack.dm [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+		heal_modifier = base_modifier
+		heal_delay = base_delay
+		organ_loss = base_organ
+
+	if(H != user)
+		if(!do_mob(user, H, heal_delay)) // 1 second delay. Heal others faster than yourself!
+			return FALSE
```
Comment (DTraitor) [inline code/modules/reagents/reagent_containers/medipack.dm:49]: This proc doesn't return anything

File: code/modules/reagents/reagent_containers/medipack.dm [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+		user.visible_message("<span class='notice'>[user] crushes the [src] over [H], the glowing goop quickly regenerating some of [H.p_their()] injuries!</span>",
+			"<span class='notice'>You squeeze the [src], and it bursts over [H], the glowing goop regenerating some of [H.p_their()] injuries.</span>")
+	else
+		if(!do_mob(user, H, (heal_delay * self_heal_delay_modifier))) // 2 second delay. Heal yourself slower than others!
+			return FALSE
```
Comment (DTraitor) [inline code/modules/reagents/reagent_containers/medipack.dm:54]: Ditto

File: code/modules/reagents/reagent_containers/medipack.dm [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+/obj/item/storage/fancy/medipack/update_overlays()
+	. = ..()
+	if(!is_open)
+		return
+
+	var/medipacks = 0
+	for(var/obj/item/medipack/large/medipack in contents)
+
+		. += image(icon = initial(icon), icon_state = medipack.in_box_sprite())
```
Comment (DTraitor) [inline code/modules/reagents/reagent_containers/medipack.dm:145]: Use `mutable_appearance()` instead

File: code/modules/reagents/reagent_containers/medipack.dm [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+/obj/item/storage/fancy/medipack/update_icon_state()
+	. = ..()
+	icon_state = "[base_icon_state]"
+
+/obj/item/storage/fancy/medipack/update_overlays()
+	. = ..()
+	if(!is_open)
+		return
+
+	var/medipacks = 0
```
Comment (DTraitor) [inline code/modules/reagents/reagent_containers/medipack.dm:142]: This var is unused

**Review by DTraitor — CHANGES_REQUESTED** [review]
Some things can be improved. Also, move the changes to the `necromorphs` folder. While sometimes we make changes to the maincode, let's avoid possible conflicts

File: code/modules/reagents/reagent_containers/medipack.dm (hunk 7 above)
Comment (francinum) [reply code/modules/reagents/reagent_containers/medipack.dm:145]: `in_box_sprite()` is only used here. Just inline it.

File: code/modules/reagents/reagent_containers/medipack.dm [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+	var/base_delay = 10
+	var/base_modifier = 0.1
+	var/base_organ = 2
+
+/obj/item/medipack/afterattack(atom/target,mob/user,prox)
+	. = ..()
+
+	if(!prox || !isliving(target))
+		return
+	var/mob/living/H = target
```
[the remark this answers is not in the document]
Comment (Ketrai) [reply code/modules/reagents/reagent_containers/medipack.dm:33]: Like !isliving(mob/living/target)
?

[the remark this answers is not in the document]
Comment (DTraitor) [reply code/modules/reagents/reagent_containers/medipack.dm:33]:
```
/obj/item/medipack/afterattack(mob/livin/target, mob/user, prox)
	. = ..()

	if(!prox || !isliving(target))
		return
```

**Pull request closed without merging**